schema/avro_builder.go

```diff
@@ -0,0 +1,91 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/hamba/avro/v2"
+)
+
+// Builder builds avro.RecordSchema instances and marshals them into JSON.
+// Builder accepts arguments for creating fields and creates them internally
+// (i.e. a user doesn't need to create the fields).
+// All errors will be returned as a joined error when marshaling the schema to JSON.
+type Builder struct {
+	errs      error
```

Comment:
A nitpick - I've been noticing the pattern in the Go community of collecting errors in `[]error` and then converting it to a single error only when needed using `errors.Join(errs...)` (for example [we do it here](https://github.com/ConduitIO/conduit-commons/blob/125d893685bf7a613c74c5c7a497aee29eb29752/config/config.go#L85)). The resulting object is slightly different, it will create a single [`joinError`](https://cs.opensource.google/go/go/+/refs/tags/go1.22.4:src/errors/join.go;l=29-31) with a single slice, otherwise we potentially have lots of nested ones.

Anyhow, it's not an important optimization, just proposing it for the sake of consistency.

Reply:
It's an easy change, so why not grab the chance?:) Thanks for letting me know about this, I wasn't aware of how it works under the hood.